### PR TITLE
Add support for openSSL

### PR DESCRIPTION
**Pull request by scaprile**

### CODE REVIEW

File: examples/mqtt-client/Makefile [hunk 1]
```diff
@@ -2,11 +2,16 @@ PROG ?= example
 MBEDTLS_DIR ?=
 
 ifeq "$(MBEDTLS_DIR)" ""
+  ifeq "$(OPENSSL)" "1"
+CFLAGS += -DMG_ENABLE_OPENSSL=1 -lssl -lcrypto
+  endif
 else
 CFLAGS += -DMG_ENABLE_MBEDTLS=1 -I$(MBEDTLS_DIR)/include -I/usr/include
 CFLAGS += -L$(MBEDTLS_DIR)/lib -lmbedtls -lmbedcrypto -lmbedx509
 endif
```
[the remark this answers is not in the document]
Comment (scaprile) [reply examples/mqtt-client/Makefile:12]: OK, if SSL=MBEDTLS do I also nested check for MBEDTLS_DIR=path or do I assume '.' ?

[the remark this answers is not in the document]
Comment (cpq) [reply examples/mqtt-client/Makefile:12]: assume all required flags are already set by CFLAGS

[the remark this answers is not in the document]
Comment (scaprile) [reply examples/mqtt-client/Makefile:7]: Got it, mbedTLS can now be installed as a standard package and there's no need for MBEDTLS_DIR anymore.

**Comment by scaprile** [comment]
I realized all jobs use SSL= so instead of being picky and calling that variable TLS I decided to just follow your suggestion and call it SSL.

**Pull request merged**